homeassistant/components/mqtt/mixins.py

```diff
@@ -487,8 +487,10 @@ async def discovery_callback(payload):
                 payload,
             )
             old_payload = self._discovery_data[ATTR_DISCOVERY_PAYLOAD]
-            debug_info.update_entity_discovery_data(self.hass, payload, self.entity_id)
-            if not payload:
+            entity_info = debug_info.update_entity_discovery_data(
+                self.hass, payload, self.entity_id
+            )
+            if not payload and entity_info is not None:
```

Comment:
What changed so we need to check `entity_info`?

Reply:
With RestoreEntity as entity class the entity is already cleaned up, no need to do this twice.

Reply:
> With RestoreEntity as entity class the entity is already cleaned up

Where does that happen, I don't see anything here: https://github.com/home-assistant/core/blob/dd5bcafab7f68ac726f199af32bd63ed25e37919/homeassistant/helpers/restore_state.py#L238-L268